### PR TITLE
added: fmt formatter for Evaluation

### PR DESCRIPTION
**Pull request by akva2**
and a simple test

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by bska** [comment]
Cool.  What's the intended usecase for this?  Is it primarily a debugging tool, or do you intend to use it for something more substantial?  The reason I ask is that we've often tried to avoid including `<fmt/format.h>` in headers, especially public headers that will be added to the installation directory during `make install`.

**Comment by akva2** [comment]
the intended use is for logging in opm-simulators, in particular killing use of stringstreams.

i can in principle encapsulate format. then we'd have to specialize the struct for 1-12 derivatives + dynamic evaluations.

**Comment by akva2** [comment]
another alternative is to move the header to opm-simulators

**Comment by bska** [comment]
> the intended use is for logging in opm-simulators, in particular killing use of stringstreams.

That's a laudable goal, of course.

> i can in principle encapsulate format. then we'd have to specialize the struct for 1-12 derivatives + dynamic evaluations.

Let's do that as follow-up work, if needed.

> another alternative is to move the header to opm-simulators

I suppose, but that seems like a hacky workaround to me.  Let's keep it here for now and see what barriers we run into.  Unless our packages already depend on libfmt, I guess it's possible that we may have to add that as a prerequisite for the next release.

File: opm/material/densead/EvaluationFormat.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+*/
+
+#ifndef OPM_EVALUATION_FORMAT_HPP
+#define OPM_EVALUATION_FORMAT_HPP
+
+#include <opm/material/densead/Evaluation.hpp>
+
+#include <fmt/format.h>
+#include <fmt/ranges.h>
+
```
Comment (blattms) [inline opm/material/densead/EvaluationFormat.hpp:31]: If this header is installed then this breaks for people that install opm-common but use the bundled fmt from there.

Comment (akva2) [reply opm/material/densead/EvaluationFormat.hpp:31]: what breaks? 

if they try to use the header, yes, but this is not (supposed to be) included anywhere libfmt is not available. in that case they'll get a compiler error.

**Pull request merged**